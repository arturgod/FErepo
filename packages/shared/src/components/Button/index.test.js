jest.mock('./layout', () => ({
  ButtonLayout: ({ children }) => <div>{children}</div>,
}));

jest.mock('./hoc/with-primary', () => ({
  withPrimary: WrappedComponent => props => (
    <WrappedComponent id="with-primary" {...props} />
  ),
}));

import React from 'react';
import { Button } from './';

describe('Button', () => {
  const props = {
    onClick: jest.fn(),
  };

  it('should exist', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should compose layout with withPrimary when primary prop is provided', () => {
    const wrapper = mount(<Button {...props} primary />);
    const withPrimary = wrapper.find('#with-primary');
    expect(withPrimary).toExist();
  });
});
