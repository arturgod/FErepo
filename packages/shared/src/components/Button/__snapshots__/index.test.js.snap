// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should exist 1`] = `
<withHandlers(branch(branch(ButtonLayout)))
  onClick={[Function]}
/>
`;
